storage/aptosdb/src/state_store/mod.rs

```diff
@@ -569,7 +572,50 @@ impl StateStore {
     ) -> Result<SparseMerkleRangeProof> {
         self.state_merkle_db.get_range_proof(rightmost_key, version)
     }
-
+    /// Put the write sets on top of current state
+    pub fn put_write_sets(
+        &mut self,
+        write_sets: Vec<WriteSet>,
+        first_version: Version,
+        ledger_batch: &SchemaBatch,
+        state_kv_batch: &SchemaBatch
+    ) -> Result<()>{
+        let _timer = OTHER_TIMERS_SECONDS
+            .with_label_values(&["update_with_writesets"])
+            .start_timer();
+        // obtain the current state
+        let current_state = self.buffered_state.lock().current_state().current.clone();
+        // get a cached state view
+        let state_view = CachedStateView::new(
+            StateViewId::Miscellaneous,
+            self.state_db.clone(),
+            first_version, // ensure the version is state_view is before the first_verson to write
+            current_state,
+            Arc::new(SyncProofFetcher::new(self.state_db.clone())),
```

Comment:
`BufferedState` and `CachedStateView` probably aren't the correct data structures to use here.. they are tools around the SMT. 

Could use things like this https://github.com/aptos-labs/aptos-core/blob/091538ffca2c55e267b08768fbf864f6ce3faacf/storage/storage-interface/src/state_view.rs#L61